### ISSUE TITLE
src/stick_table.c: assignment of overlapping memory found by coverity

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#331]
```
 926        case SMP_T_SINT:
 927                /* The stick table require a 32bit unsigned int, "sint" is a
 928                 * signed 64 it, so we can convert it inplace.
 929                 */
    3. offset: The address of *((unsigned int *)&smp->data.u.sint) is 16 bytes off of smp.
    4. offset: The address of smp->data.u.sint is 16 bytes off of smp.
    CID 1403829 (#1 of 1): Assignment of overlapping memory (OVERLAPPING_COPY)5. overlapping_assignment: Assigning smp->data.u.sint to *((unsigned int *)&smp->data.u.sint), which have overlapping memory locations and different types.
    6. target_type: *((unsigned int *)&smp->data.u.sint) has type unsigned int.
    7. source_type: smp->data.u.sint has type long long.
 930                *(unsigned int *)&smp->data.u.sint = (unsigned int)smp->data.u.sint;
 931                static_table_key.key = &smp->data.u.sint;
 932                static_table_key.key_len = 4;
 933                break;
```

**Comment by wtarreau** [comment]
Thanks Ilya. While the report is misleading (but I won't blame coverity on this one, this is really an ugly case), the code is wrong. The comment explains that it wants to extend the value but doesn't do that, it only overwrites it with itself, not changing the high bits.

**Comment by wtarreau** [comment]
OK now fixed in all affected branches (1.6 and onwards).
